models/users.js

```diff
@@ -572,6 +572,38 @@ const fetchAllUsers = async () => {
   return users;
 };
 
+const archiveUserIfNotInDiscord = async () => {
+  try {
+    const snapshot = await userModel.where("roles.in_discord", "==", false).get();
+    const batch = firestore.batch();
+    const summary = {
+      totalUsersWithArchivedRoleUpdated: 0,
+    };
+
+    snapshot.forEach((user) => {
+      const id = user.id;
+      const userData = user.data();
+
+      const updatedUserData = {
+        ...userData,
+        roles: {
+          ...userData.roles,
+          archived: true,
+        },
+      };
```

Comment:
Before updating please check if the `archived` role is false or not. if its already true we need not update.